### PR TITLE
FIX: correctly destroy menu/tooltip when removed from dom

### PR DESCRIPTION
**Pull request by jjaffeux**
This fix will ensure we correctly play the close/destroy sequence in this situation.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/bookmark-menu.gjs [hunk 1]
```diff
@@ -2,7 +2,6 @@ import Component from "@glimmer/component";
 import { tracked } from "@glimmer/tracking";
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/components/bookmark-menu.gjs:2]: Not sure about this one, it did appear in some retained trace and that looked suspicious, given I think the new code is just better, I didn't spend much more time thinking about it

File: app/assets/javascripts/float-kit/addon/components/d-float-body.gjs [hunk 2]
```diff
@@ -52,7 +52,7 @@ export default class DFloatBody extends Component {
   <template>
     <DFloatPortal
       @inline={{@inline}}
-      @portalOutletElement={{@instance.portalOutletElement}}
+      @portalOutletElement={{(@instance.portalOutletElement)}}
```
Comment (jjaffeux) [inline app/assets/javascripts/float-kit/addon/components/d-float-body.gjs:55]: Im also not sure about this one, but that feels more correct, having a property on the object with a reference to a DOM node is always a potential leak, using a function to grab it when we need it avoids this risk I think.

File: app/assets/javascripts/float-kit/addon/components/d-menu.gjs [hunk 3]
```diff
@@ -25,6 +25,11 @@ export default class DMenu extends Component {
   registerTrigger = modifier((element) => {
     this.menuInstance.trigger = element;
     this.options.onRegisterApi?.(this.menuInstance);
+
+    return () => {
+      this.menuInstance.destroy();
+      this.menuInstance = null;
+    };
```
Comment (jjaffeux) [inline app/assets/javascripts/float-kit/addon/components/d-menu.gjs:32]: I noticed we were not correctly calling destroy when the menu was removed from DOM.
Im not sure we need to nullify it, but it seems like an appropriate time to do it.

File: app/assets/javascripts/float-kit/addon/lib/d-toast-instance.js [hunk 4]
```diff
@@ -13,7 +13,11 @@ export default class DToastInstance {
 
   constructor(owner, options = {}) {
     setOwner(this, owner);
-    this.options = { ...TOAST.options, ...options };
+    this.options = {
+      ...TOAST.options,
+      ...{ component: DDefaultToast },
```
Comment (jjaffeux) [inline app/assets/javascripts/float-kit/addon/lib/d-toast-instance.js:18]: This one also appeared in a retained trace, Im not sure important a component and storing it in an array was a great idea. It was probably not causing much harm though.

File: app/assets/javascripts/float-kit/addon/services/menu.js [hunk 5]
```diff
@@ -2,11 +2,11 @@ import { action } from "@ember/object";
 import { getOwner } from "@ember/owner";
 import { schedule } from "@ember/runloop";
 import Service from "@ember/service";
-import { TrackedSet } from "@ember-compat/tracked-built-ins";
+import { TrackedObject } from "@ember-compat/tracked-built-ins";
```
Comment (jjaffeux) [inline app/assets/javascripts/float-kit/addon/services/menu.js:5]: All the fixes around registeredMenus and registeredTooltips were legit I think. Basically I discovered that:

```javascript
set.add(instance);
set.has(instance); // false
```

**Review by davidtaylorhq — APPROVED** [review]

**Pull request merged**